subsys/bluetooth/audio/shell/bap.c

```diff
@@ -3649,47 +3654,75 @@ static int cmd_init(const struct shell *sh, size_t argc, char *argv[])
 		return -ENOEXEC;
 	}
 
+	if (argc != 1 && argc != 3) {
+		shell_error(sh, "Invalid argument count");
+		shell_help(sh);
+	}
+
 #if defined(CONFIG_BT_BAP_UNICAST_SERVER)
+	unsigned long snk_cnt, src_cnt;
+
+	if (argc == 3) {
+		snk_cnt = shell_strtoul(argv[1], 0, &err);
+		if (snk_cnt > CONFIG_BT_ASCS_MAX_ASE_SNK_COUNT) {
+			shell_error(sh, "Invalid Sink ASE count: %lu. Valid interval: [0, %u]",
+				snk_cnt, CONFIG_BT_ASCS_MAX_ASE_SNK_COUNT);
+
+			return -ENOEXEC;
+		}
+
+		unicast_server_param.snk_cnt = snk_cnt;
+
+		src_cnt = shell_strtoul(argv[2], 0, &err);
+		if (src_cnt > CONFIG_BT_ASCS_MAX_ASE_SRC_COUNT) {
+			shell_error(sh, "Invalid Source ASE count: %lu. Valid interval: [0, %u]",
+				src_cnt, CONFIG_BT_ASCS_MAX_ASE_SRC_COUNT);
+
+			return -ENOEXEC;
+		}
+
+		unicast_server_param.src_cnt = src_cnt;
+	} else {
+		snk_cnt = CONFIG_BT_ASCS_MAX_ASE_SNK_COUNT;
+		src_cnt = CONFIG_BT_ASCS_MAX_ASE_SRC_COUNT;
+	}
+
+	bt_bap_unicast_server_register(&unicast_server_param);
 	bt_bap_unicast_server_register_cb(&unicast_server_cb);
 #endif /* CONFIG_BT_BAP_UNICAST_SERVER */
 
 #if defined(CONFIG_BT_PAC_SNK)
-	bt_pacs_cap_register(BT_AUDIO_DIR_SINK, &cap_sink);
+		bt_pacs_cap_register(BT_AUDIO_DIR_SINK, &cap_sink);
 #endif /* CONFIG_BT_PAC_SNK */
 #if defined(CONFIG_BT_PAC_SRC)
-	bt_pacs_cap_register(BT_AUDIO_DIR_SOURCE, &cap_source);
+		bt_pacs_cap_register(BT_AUDIO_DIR_SOURCE, &cap_source);
 #endif /* CONFIG_BT_PAC_SNK */
 
-	if (IS_ENABLED(CONFIG_BT_PAC_SNK_LOC)) {
-		err = bt_pacs_set_location(BT_AUDIO_DIR_SINK, LOCATION);
-		__ASSERT(err == 0, "Failed to set sink location: %d", err);
+	err = bt_pacs_set_location(BT_AUDIO_DIR_SINK, LOCATION);
+	__ASSERT(err == 0, "Failed to set sink location: %d", err);
+	err = bt_pacs_set_supported_contexts(BT_AUDIO_DIR_SINK,
+					     CONTEXT);
+	__ASSERT(err == 0, "Failed to set sink supported contexts: %d",
+		 err);
+	err = bt_pacs_set_available_contexts(BT_AUDIO_DIR_SINK,
+					     CONTEXT);
+	__ASSERT(err == 0, "Failed to set sink available contexts: %d",
+		 err);
 
-		err = bt_pacs_set_supported_contexts(BT_AUDIO_DIR_SINK,
-						     CONTEXT);
-		__ASSERT(err == 0, "Failed to set sink supported contexts: %d",
-			 err);
+	err = bt_pacs_set_location(BT_AUDIO_DIR_SOURCE, LOCATION);
+	__ASSERT(err == 0, "Failed to set source location: %d", err);
 
-		err = bt_pacs_set_available_contexts(BT_AUDIO_DIR_SINK,
-						     CONTEXT);
-		__ASSERT(err == 0, "Failed to set sink available contexts: %d",
-			 err);
-	}
+	err = bt_pacs_set_supported_contexts(BT_AUDIO_DIR_SOURCE,
+					     CONTEXT);
+	__ASSERT(err == 0, "Failed to set sink supported contexts: %d",
+		 err);
 
-	if (IS_ENABLED(CONFIG_BT_PAC_SRC_LOC)) {
-		err = bt_pacs_set_location(BT_AUDIO_DIR_SOURCE, LOCATION);
-		__ASSERT(err == 0, "Failed to set source location: %d", err);
+	err = bt_pacs_set_available_contexts(BT_AUDIO_DIR_SOURCE,
+					     CONTEXT);
+	__ASSERT(err == 0,
+		 "Failed to set source available contexts: %d",
+		 err);
```

Comment:
Why was the `if (IS_ENABLED(CONFIG_BT_PAC_SNK_LOC)) {` and `if IS_ENABLED(CONFIG_BT_PAC_SRC_LOC)) {` checks removed?